### PR TITLE
internal/config: add DD_APM_ prefixed env. vars duplicates

### PR DESCRIPTION
**Pull request by gbbr**
Adds new environment variables (correctly prefixed), marking others as deprecated:
* DD_APM_CONNECTION_LIMIT
* DD_APM_RECEIVER_PORT
* DD_APM_MAX_EPS
* DD_APM_MAX_TPS
* DD_APM_IGNORE_RESOURCES

### CODE REVIEW

File: internal/config/env.go [hunk 1]
```diff
@@ -14,32 +14,44 @@ func applyEnv() {
...
 	// when using the legacy config loader.
 	for envKey, cfgKey := range map[string]string{
-		"DD_SITE":                  "site",
-		"DD_API_KEY":               "api_key",
-		"DD_HOSTNAME":              "hostname",
-		"DD_BIND_HOST":             "bind_host",
-		"DD_DOGSTATSD_PORT":        "dogstatsd_port",
-		"DD_LOG_LEVEL":             "log_level",
-		"DD_CONNECTION_LIMIT":      "apm_config.connection_limit",
+		// Core agent:
```
Comment (erichgess) [inline internal/config/env.go]: It's possible that a customer will have both the deprecated and the current env var name defined, so we should have an explicit order of precedence for deprecated env var names and current env var names.  So if both the deprecated and the current name for a parameter is defined we always use the current one.  

However, ordering on a map is not guaranteed so this loop may prioritize deprecated names over current names.  

Can we change this to support ensuring that if both versions of a name are defined we'll always get the current one?

Comment (gbbr) [reply internal/config/env.go]: Nice catch! You’re totally right about map ordering. Will change this.

Comment (erichgess) [reply internal/config/env.go]: No problem.  Glad my Golang skills are getting good enough to be helpful!

**Comment by gbbr** [comment]
@erichgess I've added specific tests for this and replaced the map with an ordered sequence (a slice). PTAL

File: internal/config/env.go (hunk 1 above)
Comment (gbbr) [reply internal/config/env.go]: Done!

File: internal/config/env_test.go [hunk 2]
```diff
@@ -14,6 +15,35 @@ func TestMain(m *testing.M) {
 }
 
 func TestLoadEnv(t *testing.T) {
+	t.Run("overrides", func(t *testing.T) {
+		// tests that newer envs. override deprecated ones
```
Comment (erichgess) [inline internal/config/env_test.go]: 👍

**Pull request merged**